### PR TITLE
Add .Net Framework versions to 4.7.2, 4.8 to Legacy and update deps

### PR DESCRIPTION
**Pull request by jussivesa**
Add .Net Framework versions to 4.7.2, 4.8 and update deps
- Drop net461 support from non-Legacy
- Add net48 support for non-Legacy
- Add net472 and net48 support for Legacy
- Add net472 to net48 for non-Legacy
- Update deps based on targeting framework. When applicant, updates:
  - Newtonsoft.Json 10.0.3 -> 13.0.1
  - System.Reactive 3.1.1 -> 5.0.0
- Shared dep updates as following:
  - FluentAssertions 4.19.4 -> 5.1.2
  - Microsoft.NET.Test.Sdk 15.3.0-preview -> 15.6.0
  - Microsoft.NETFramework.ReferenceAssemblies 1.0.0 -> 1.0.2

This will drop net461 support, and we will publish RxMq version 2.0.0 from this as this is a breaking change.

### CODE REVIEW

File: Create-ClientSecrets.ps1 [hunk 1]
```diff
@@ -25,5 +25,10 @@ $output = @{
     AzureMaximumRetryCount   = 10
 }
 
-$output | ConvertTo-Json -depth 100 | Out-File "Protacon.RxMq.AzureServiceBus.Tests\client-secrets.json"
-Copy-Item "Protacon.RxMq.AzureServiceBus.Tests\client-secrets.json" -Destination "Protacon.RxMq.AzureServiceBusLegacy.Tests"
+$filename = "client-secrets.json"
+$projectLocation = "Protacon.RxMq.AzureServiceBus.Tests"
+$legacyProjectLocation = "Protacon.RxMq.AzureServiceBusLegacy.Tests"
```
Comment (Lem82) [inline Create-ClientSecrets.ps1:30]: Pitäisikö nimen olla ProjectTestsLocation ja legacyProjectTestsLocation

File: Protacon.RxMq.AzureServiceBusLegacy/packagerefs.net48.targets [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿<Project>
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="System.Reactive" Version="5.0.0" />
```
Comment (Lem82) [inline Protacon.RxMq.AzureServiceBusLegacy/packagerefs.net48.targets:4]: Mihin tätä tarvitaan?

**Review by Lem82 — APPROVED** [review]
Nice ! Only few questions

File: Create-ClientSecrets.ps1 (hunk 1 above)
Comment (jussivesa) [reply Create-ClientSecrets.ps1:30]: Could but not sure if it's worth to rename

File: Protacon.RxMq.AzureServiceBusLegacy/packagerefs.net48.targets (hunk 2 above)
Comment (jussivesa) [reply Protacon.RxMq.AzureServiceBusLegacy/packagerefs.net48.targets:4]: Targeting net48, we want to have latest Newtonsoft and it depends on System.Reactive, hence the updates.

**Pull request merged**